drivers/sensor/lis2mdl/Kconfig

```diff
@@ -11,6 +11,20 @@ menuconfig LIS2MDL
 
 if LIS2MDL
 
+choice LIS2MDL_OPERATION_MODE
+	prompt "Operation mode"
+	default LIS2MDL_CONTINUOUS_MODE
+	help
+	  Specify the type of operation mode to be used by driver.
+
+config LIS2MDL_SINGLE_MODE
+	bool "Single mode"
```

Comment:
Is it true that in single case mode the trigger should be put to on?
I saw that you are giving a semaphore inside the trigger.c module

Reply:
If you mean lis2mdl_trigger_set() function by trigger? The answer is NO. 
But we need an interrupt handler to give the semaphore I used for blocking the fetch function. 

Therefore I used "k_sem_give(&lis2mdl->fetch_sem);" in the lis2mdl_handle_interrupt(). The lis2mdl_handle_interrupt() function will be enabled by choosing the CONFIG_LIS2MDL_TRIGGER_OWN_THREAD.

So in my implementation, the user needs to choose both the **LIS2MDL_SINGLE_MODE** and **CONFIG_LIS2MDL_TRIGGER_OWN_THREAD** to have everything working.